### PR TITLE
Relativize cloud base links

### LINKED ISSUE #3043: Relativize cloud links in API generation script
When we switch packages' intersphinx config to point to https://quantum.cloud.ibm.com/, it'll include links to that website rather than the legacy website. We should relativize those links in the same way we do the legacy links.

Here's an example of generating the runtime dev docs with the new `objects.inv` location:
Image

### PR DESCRIPTION
**Pull request by frankharkins**
Closes #3043

### CODE REVIEW

File: scripts/js/lib/api/updateLinks.ts [hunk 1]
```diff
@@ -166,7 +172,8 @@ export function relativizeLink(link: Link): Link | undefined {
 
   const newText = link.url === link.text ? url : undefined;
   const newPrefix = priorPrefixToNewPrefix.get(priorPrefix)!;
-  return { url: `${newPrefix}/${url}`, text: newText };
+  const relativeUrl = removePrefix(join(newPrefix, url), "/");
+  return { url: `/${relativeUrl}`, text: newText };
```
Comment (frankharkins) [inline scripts/js/lib/api/updateLinks.ts:173]: I had to change the logic here to fix some cases of double-slashes (e.g. `/docs//guides/page`).

Comment (Eric-Arellano) [reply scripts/js/lib/api/updateLinks.ts:173]: Sg. Do your tests capture that edge case?

Comment (frankharkins) [reply scripts/js/lib/api/updateLinks.ts:173]: Yep, the tests actually uncovered it
